src/Common/src/Interop/Unix/System.Native/Interop.GetUnixVersion.cs

```diff
@@ -11,26 +12,39 @@ internal static partial class Interop
     internal static partial class Sys
     {
         [DllImport(Libraries.SystemNative, EntryPoint = "SystemNative_GetUnixVersion", CharSet = CharSet.Ansi, SetLastError = true)]
-        private static extern int GetUnixVersion(StringBuilder version, out int capacity);
+        private static extern int GetUnixVersion(sbyte[] version, ref int capacity);
 
         internal static string GetUnixVersion()
         {
             // max value of _UTSNAME_LENGTH on known Unix platforms is 1024.
             const int _UTSNAME_LENGTH = 1024;
-            int capacity = _UTSNAME_LENGTH * 3 + 2;
-            StringBuilder version = new StringBuilder(capacity);
+            int capacity = _UTSNAME_LENGTH + 1; // +1 for null terminator
+            var version = new sbyte[capacity];
 
-            if (GetUnixVersion(version, out capacity) != 0)
+            bool success = GetUnixVersion(version, ref capacity) == 0;
+            if (!success)
             {
                 // Check if the function failed due to insufficient buffer.
-                if (capacity > version.Capacity)
+                if (capacity > version.Length)
                 {
-                    version.Capacity = capacity;
-                    GetUnixVersion(version, out capacity);
+                    version = new sbyte[capacity];
+                    success = GetUnixVersion(version, ref capacity) == 0;
                 }
             }
 
-            return version.ToString();
+            if (!success)
+            {
+                return string.Empty;
+            }
+
+            Debug.Assert(Array.IndexOf(version, 0) != -1);
+            unsafe
+            {
+                fixed (sbyte* ptr = version)
```

Comment:
Nit: I would use `byte` in the whole function, and cast it to `(sbyte*)` just to call the constructor. I find the convention used by the few constructors to use `sbyte` for "ansi" chars pretty odd.